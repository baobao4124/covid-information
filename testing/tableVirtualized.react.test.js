'use strict';

import React from 'react';
import MainTable from 'Layouts/AppMain/components/TableVirtualized/components/Main/index';
import FooterTable from 'Layouts/AppMain/components/TableVirtualized/components/Footer';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(<MainTable />).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders footer correctly', () => {
    const tree = renderer.create(<FooterTable />).toJSON();

    expect(tree).toMatchSnapshot();
});