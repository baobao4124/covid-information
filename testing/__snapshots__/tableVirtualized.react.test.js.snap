// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders footer correctly 1`] = `
<div
  className="wrap-table-footer"
  style={
    Object {
      "backgroundColor": "white",
      "border": "1px solid #e3eef1",
    }
  }
>
  <div
    className="record-table"
  >
    <span
      className="record-name"
      style={
        Object {
          "marginLeft": "30px",
        }
      }
    >
      Go to page
    </span>
    <input
      onChange={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "textAlign": "center",
          "width": "30px",
        }
      }
      type="text"
      value={1}
    />
    <span
      className="record-name"
    >
      Show rows
    </span>
    <div
      className="d-inline-block dropdown"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        aria-label={null}
        className="btn-full dropdown-toggle btn btn-default"
        onClick={[Function]}
        type="button"
      >
        10
      </button>
      <div
        aria-hidden={true}
        className="dropdown-menu-rows dropdown-menu"
        role="menu"
        tabIndex="-1"
      >
        <button
          className="dropdown-menu-rows-item active dropdown-item"
          onClick={[Function]}
          role="menuitem"
          tabIndex="0"
          type="button"
        >
          10
        </button>
        <button
          className="dropdown-menu-rows-item dropdown-item"
          onClick={[Function]}
          role="menuitem"
          tabIndex="0"
          type="button"
        >
          30
        </button>
        <button
          className="dropdown-menu-rows-item dropdown-item"
          onClick={[Function]}
          role="menuitem"
          tabIndex="0"
          type="button"
        >
          50
        </button>
        <button
          className="dropdown-menu-rows-item dropdown-item"
          onClick={[Function]}
          role="menuitem"
          tabIndex="0"
          type="button"
        >
          100
        </button>
        <button
          className="dropdown-menu-rows-item dropdown-item"
          onClick={[Function]}
          role="menuitem"
          tabIndex="0"
          type="button"
        >
          200
        </button>
        <button
          className="dropdown-menu-rows-item dropdown-item"
          onClick={[Function]}
          role="menuitem"
          tabIndex="0"
          type="button"
        >
          500
        </button>
      </div>
    </div>
    <span
      className="record-num"
    >
      1 - 10 of 100
    </span>
    <div
      className="d-inline-block"
    >
      <ul
        className="pagination"
      >
        <li
          className="page-item disabled first"
        >
          <a
            className="page-link"
          >
            <i
              className="icon-ants-step-backward "
              role="icon"
              style={Object {}}
              type="icon-ants-step-backward"
            />
          </a>
        </li>
        <li
          className="page-item disabled prev"
        >
          <a
            className="page-link"
          >
            <i
              className="icon-ants-caret-left "
              role="icon"
              style={Object {}}
              type="icon-ants-caret-left"
            />
          </a>
        </li>
        <li
          className="page-item active"
        >
          <a
            className="page-link"
            onClick={[Function]}
          >
            1
          </a>
        </li>
        <li
          className="page-item false"
        >
          <a
            className="page-link"
            onClick={[Function]}
          >
            2
          </a>
        </li>
        <li
          className="page-item false"
        >
          <a
            className="page-link"
            onClick={[Function]}
          >
            3
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            onClick={[Function]}
          >
            ...
          </a>
        </li>
        <li
          className="page-item false next"
          onClick={[Function]}
        >
          <a
            className="page-link"
          >
            <i
              className="icon-ants-caret-right "
              role="icon"
              style={Object {}}
              type="icon-ants-caret-right"
            />
          </a>
        </li>
        <li
          className="page-item false last"
          onClick={[Function]}
        >
          <a
            className="page-link"
          >
            <i
              className="icon-ants-step-forward "
              role="icon"
              style={Object {}}
              type="icon-ants-step-forward"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
